doc: Adds migration guides to transition out of Serverless and Shared-tier clusters (#2746)
mongodb/terraform-provider-mongodbatlas: docs/guides/1.22.0-upgrade-guide.md

---
page_title: "Upgrade Guide 1.22.0"
---

# MongoDB Atlas Provider 1.22.0: Upgrade and Information Guide

The Terraform MongoDB Atlas Provider version 1.22.0 has a number of new and exciting features.

## New Resources, Data Sources, and Features

- You can now manage Flex Clusters with the new `mongodbatlas_flex_cluster` resource and corresponding data sources. The feature is available as a preview feature. To learn more, please review `mongodbatlas_flex_cluster` [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/flex_cluster).

## Deprecations and removals

> - `continuous_backup_enabled` attribute has been deprecated in `mongodbatlas_serverless_instance` resource and data sources. If your workload needs this feature, we recommend to switch to Atlas Dedicated clusters instead. For more details please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).

Reviewer: ```suggestion
- `continuous_backup_enabled` attribute is deprecated in `mongodbatlas_serverless_instance` resource and data sources. If your workload requires this feature, we recommend switching to Atlas Dedicated clusters. To learn more, see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).
```

1. Prefer the present tense, where possible.
2. 'requires' is a bit more appropriate than 'needs' in this case; 'needs' carries a connotation of agency, and while there are specific use cases for it with an inanimate thing (workflow), it shouldn't be used generically as it can with animate things.
3. 'instead' can be paired with 'use' as in 'use X instead', but 'switch to' already implies a relationship of alternatives, making 'instead' redundant.
4. We are gradually standardizing on 'to learn more', or else 'to learn more about ____' in these situations. Please apply similar adjustments to the other places where this applies.

> - `auto_indexing` attribute has been deprecated in `mongodbatlas_serverless_instance` resource and data sources. If your workload needs this feature, we recommend to switch to Atlas Dedicated clusters instead. For more details please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).

Reviewer: Same comments as above.

> - `mongodbatlas_privatelink_endpoint_serverless` resource has been deprecated. If your workload requires this feature, we recommend switching to Atlas Dedicated clusters. To learn more, please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).

Reviewer: ```suggestion
- `mongodbatlas_privatelink_endpoint_serverless` resource is deprecated. If your workload requires this feature, we recommend switching to Atlas Dedicated clusters. To learn more, please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).
```

Favor active voice.

> - `mongodbatlas_privatelink_endpoint_serverless` resource has been deprecated. If your workload needs this feature, we recommend to switch to Atlas Dedicated clusters instead. For more details please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).

Reviewer: Same comments as above.

> - `mongodbatlas_privatelink_endpoint_service_serverless` resource, `mongodbatlas_privatelink_endpoint_service_serverless` and `mongodbatlas_privatelink_endpoints_service_serverless` data sources have been deprecated. If your workload requires this feature, we recommend switching to Atlas Dedicated clusters. To learn more, please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).

Reviewer: ```suggestion
- `mongodbatlas_privatelink_endpoint_service_serverless` resource, `mongodbatlas_privatelink_endpoint_service_serverless` and `mongodbatlas_privatelink_endpoints_service_serverless` data sources are deprecated. If your workload requires this feature, we recommend switching to Atlas Dedicated clusters. To learn more, please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).
```
Favor active voice.

> - `mongodbatlas_privatelink_endpoint_service_serverless` resource, `mongodbatlas_privatelink_endpoint_service_serverless` and `mongodbatlas_privatelink_endpoints_service_serverless` data sources have been deprecated.If your workload needs this feature, we recommend to switch to Atlas Dedicated clusters instead. For more details please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).

Reviewer: ```suggestion
- `mongodbatlas_privatelink_endpoint_service_serverless` resource, `mongodbatlas_privatelink_endpoint_service_serverless` and `mongodbatlas_privatelink_endpoints_service_serverless` data sources have been deprecated. If your workload needs this feature, we recommend to switch to Atlas Dedicated clusters instead. For more details please see the [Migration Guide: Transition out of Serverless Instances and Shared-tier clusters](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/serverless-shared-migration-guide).
```

Add a space between the period and the next word. Otherwise, same comments as above.
